From c7181e7d4953c9e4e1152eb2dcbd8ed54c38e405 Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Tue, 9 May 2023 14:19:11 +0800
Subject: [PATCH 395/780] [Huawei] blk-mq: export blk-mq-debugfs apis

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188826

--------------------------------

These apis will be used later to create new debugfs files. There are no
functional changes.

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
---
 block/blk-mq-debugfs.c | 10 ++--------
 block/blk-mq-debugfs.h |  9 +++++++++
 2 files changed, 11 insertions(+), 8 deletions(-)

diff --git a/block/blk-mq-debugfs.c b/block/blk-mq-debugfs.c
index 4544bc60a527..a63aef927a9c 100644
--- a/block/blk-mq-debugfs.c
+++ b/block/blk-mq-debugfs.c
@@ -846,8 +846,8 @@ static const struct blk_mq_debugfs_attr blk_mq_debugfs_ctx_attrs[] = {
 	{},
 };
 
-static void debugfs_create_files(struct dentry *parent, void *data,
-				 const struct blk_mq_debugfs_attr *attr)
+void debugfs_create_files(struct dentry *parent, void *data,
+			  const struct blk_mq_debugfs_attr *attr)
 {
 	if (IS_ERR_OR_NULL(parent))
 		return;
@@ -859,12 +859,6 @@ static void debugfs_create_files(struct dentry *parent, void *data,
 				    (void *)attr, &blk_mq_debugfs_fops);
 }
 
-static bool blk_mq_debugfs_enabled(struct request_queue *q)
-{
-	return !IS_ERR_OR_NULL(q->debugfs_dir) &&
-	       test_bit(QUEUE_FLAG_DEBUGFS, &q->queue_flags);
-}
-
 void blk_mq_debugfs_register(struct request_queue *q)
 {
 	struct blk_mq_hw_ctx *hctx;
diff --git a/block/blk-mq-debugfs.h b/block/blk-mq-debugfs.h
index 3a2c43a9a0ae..b2b2057f3acf 100644
--- a/block/blk-mq-debugfs.h
+++ b/block/blk-mq-debugfs.h
@@ -34,6 +34,15 @@ void blk_mq_debugfs_unregister_sched_hctx(struct blk_mq_hw_ctx *hctx);
 
 void blk_mq_debugfs_register_rqos(struct rq_qos *rqos);
 void blk_mq_debugfs_unregister_rqos(struct rq_qos *rqos);
+void debugfs_create_files(struct dentry *parent, void *data,
+			  const struct blk_mq_debugfs_attr *attr);
+
+static inline bool blk_mq_debugfs_enabled(struct request_queue *q)
+{
+	return !IS_ERR_OR_NULL(q->debugfs_dir) &&
+	       test_bit(QUEUE_FLAG_DEBUGFS, &q->queue_flags);
+}
+
 #else
 static inline void blk_mq_debugfs_register(struct request_queue *q)
 {
-- 
2.17.1

